storage/database.go

```diff
@@ -551,10 +554,13 @@ func (d *db) ownedNamespacesWithLock() []databaseNamespace {
 	return namespaces
 }
 
-func (d *db) GetOwnedNamespaces() []databaseNamespace {
+func (d *db) GetOwnedNamespaces() ([]databaseNamespace, error) {
```

Comment:
could you add a test to verify this functions as you're expecting

Reply:
yep done